miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/StorageLocationRestController.java

```diff
@@ -306,10 +305,8 @@ public void bulkDelete(@RequestBody(required = true) List<Long> ids) throws IOEx
   public @ResponseBody ServiceRecordDto createRecord(
       @PathVariable(name = "locationId", required = true) Long locationId, @RequestBody ServiceRecordDto dto)
       throws IOException {
-    StorageLocation retrievedFreezer = RestUtils.retrieve("Storage location", locationId, storageLocationService);
-    if (retrievedFreezer == null) {
-      throw new NotFoundException("No freezer found for ID " + locationId.toString());
-    }
+    RestUtils.retrieve("Storage location", locationId, storageLocationService);
+
     ServiceRecord record = Dtos.to(dto);
     StorageLocation freezer = storageLocationService.get(locationId);
```

Comment:
Freezer is being fetched twice. Use `RestUtils#retrieve` only to get the freezer